src/utils/SignalingSpeakingHandler.js

```diff
@@ -0,0 +1,127 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * Helper to handle signaling messages for speaking / speakingStopped notifications.
+ *
+ * The store is updated when local or remote participants change their speaking status.
+ *
+ * It is expected that the speaking status of participant will be
+ * modified only when the current conversation is joined and call is started.
+ *
+ */
+export default class SignalingSpeakingHandler {
+
+	// Constants, properties
+	#store
+	#localMediaModel
+	#callParticipantCollection
+
+	// Methods (bounded to have access to 'this')
+	#handleAddParticipantBound
+	#handleRemoveParticipantBound
+	#handleLocalSpeakingBound
+	#handleSpeakingBound
+
+	constructor(store, localMediaModel, callParticipantCollection) {
+		this.#store = store
+		this.#localMediaModel = localMediaModel
+		this.#callParticipantCollection = callParticipantCollection
+
+		this.#handleAddParticipantBound = this.#handleAddParticipant.bind(this)
+		this.#handleRemoveParticipantBound = this.#handleRemoveParticipant.bind(this)
+		this.#handleLocalSpeakingBound = this.#handleLocalSpeaking.bind(this)
+		this.#handleSpeakingBound = this.#handleSpeaking.bind(this)
+
+		this.#localMediaModel.on('change:speaking', this.#handleLocalSpeakingBound)
+		this.#localMediaModel.on('change:stoppedSpeaking', this.#handleLocalSpeakingBound)
+
+		this.#callParticipantCollection.on('add', this.#handleAddParticipantBound)
+		this.#callParticipantCollection.on('remove', this.#handleRemoveParticipantBound)
+	}
+
+	/**
+	 * Destroy a handler, remove all listeners, purge the speaking state from store
+	 *
+	 */
+	destroy() {
+		this.#localMediaModel.off('change:speaking', this.#handleLocalSpeakingBound)
+		this.#localMediaModel.off('change:stoppedSpeaking', this.#handleLocalSpeakingBound)
+
+		this.#callParticipantCollection.off('add', this.#handleAddParticipantBound)
+		this.#callParticipantCollection.off('remove', this.#handleRemoveParticipantBound)
+
+		this.#store.dispatch('purgeSpeakingStore', { token: this.#store.getters.getToken() })
+	}
+
+	/**
+	 * Add listeners for speaking signals on added participants model
+	 *
+	 * @param {object} callParticipantCollection the collection of external participant models
+	 * @param {object} callParticipantModel the added participant model
+	 */
+	#handleAddParticipant(callParticipantCollection, callParticipantModel) {
+		callParticipantModel.on('change:speaking', this.#handleSpeakingBound)
+		callParticipantModel.on('change:stoppedSpeaking', this.#handleSpeakingBound)
+	}
+
+	/**
+	 * Remove listeners for speaking signals on removed participants model
```

Comment:
As in the class description, it should be something like _speaking status changes_ rather than _speaking signals_ (unless here signal refers to events).